File: day49/a1.py
from decimal import Decimal, ROUND_HALF_UP


wrong_price = 19.99
wrong_total = wrong_price * 2.9

price = Decimal("19.99")
quantity = Decimal("2.9")
correct_total = price * quantity


print(f"Total: ${correct_total}")

tax =  total * Decimal("0.12")
tax = tax.quantize(Decimal("0.01"), ROUND_HALF_UP)

print(f"Tax (12%): ${tax}")

File: day49/a1.java
import java.math.BigDecimal;
import java.math.RoundingMode;
public class a1{
    public static void main(String[] args){

        double wrongPrice = 19.99;
        double wrongTotal = wrongPrice * 2.9;


        BigDecimal price = new BigDecimal("19.99");
        BigDecimal quantity = new BigDecimal("2.9");
        BigDecimal correctTotal = price.multiply(quantity);


        System.out.println("Total: $" + correctTotal);

        BigDecimal tax = correctTotal.multiply(new BigDecimal("0.12"));
        tax = tax.setScale(2,RoundingMode.HALF_UP);
        System.out.println("Tax (12%): $" + tax);
    }
}